expr/cast.go

```diff
@@ -0,0 +1,144 @@
+package expr
+
+import (
+	"math"
+	"net"
+
+	"github.com/brimsec/zq/expr/coerce"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zng"
+)
+
+type ValueCaster func(zv zng.Value) (zng.Value, error)
+
+func LookupValueCaster(typ string) ValueCaster {
+	switch typ {
+	case "int8":
+		return castToInt8
+	case "int16":
+		return castToInt16
+	case "int32":
+		return castToInt32
+	case "int64":
+		return castToInt64
+	case "uint8":
+		return castToUint8
+	case "uint16":
+		return castToUint16
+	case "uint32":
+		return castToUint32
+	case "uint64":
+		return castToUint64
+	case "float64":
+		return castToFloat64
+	case "ip":
+		return castToIP
+	case "time":
+		return castToTime
+	case "string":
+		return castToString
+	case "bstring":
+		return castToBstring
+	case "bytes":
+		return castToBytes
+	default:
+		return nil
+	}
+}
+
+func castToIntN(min, max int64, typ zng.Type) func(zng.Value) (zng.Value, error) {
```

Comment:
Nit: `typ` feels more prominent than `min` and `max`, so I'd make this this `castToIntN(typ zng.Type, min, max int64)`.

Reply:
👍